src/systems/systems/src/hydraulic/linear_actuator.rs

```diff
@@ -0,0 +1,1302 @@
+use nalgebra::{Rotation3, Unit, Vector3};
+
+use uom::si::{
+    angle::radian,
+    angular_acceleration::radian_per_second_squared,
+    angular_velocity::radian_per_second,
+    area::square_meter,
+    f64::*,
+    force::newton,
+    length::meter,
+    mass::kilogram,
+    ratio::ratio,
+    torque::newton_meter,
+    velocity::meter_per_second,
+    volume::{cubic_meter, gallon},
+    volume_rate::gallon_per_second,
+};
+
+use crate::simulation::UpdateContext;
+
+pub trait Actuator {
+    fn used_volume(&self) -> Volume;
+    fn reservoir_return(&self) -> Volume;
+    fn reset_accumulators(&mut self);
+}
+
+/// Trait linked to anything moving bounded between a minimum and maximum position.
+/// Those bounds implies that it will have a max and min distance from a fixed
+/// point in space to the position where we apply forces to it,
+/// for example a control arm where an actuator is attached.
+pub trait BoundedLinearLength {
+    fn min_absolute_length_to_anchor(&self) -> Length;
+    fn max_absolute_length_to_anchor(&self) -> Length;
+}
+
+#[derive(PartialEq, Clone, Copy)]
+pub enum LinearActuatorMode {
+    ClosedValves,
+    PositionControl,
+    ActiveDamping,
+}
+
+/// Represents an abstraction of the low level hydraulic actuator control system that would in real life consist of a lot of
+/// solenoid control valves, spring loaded valves, differential pressure mechanism......
+///
+/// We don't want to simulate all of those little bits, so the functions of the actuator are split in
+/// functional modes
+///
+/// ClosedValves -> Turns actuator in a high constant spring/damper system simulating a closed actuator
+/// only constrained by its own fluid compressibility
+///
+/// ActiveDamping -> Actuator use internal valves to provide a force resisting to its own movements, dampening
+/// the piece movements it's connected to
+///
+/// PositionControl -> Actuator will try to use hydraulic pressure to move to a requested position, while
+/// maintaining flow limitations
```

Comment:
**suggestion**: Rust docs support markdown, using it makes things clearer.
**suggestion**: Consistent usage of `.`.

```suggestion
/// Represents an abstraction of the low level hydraulic actuator control system that would in real life consist of a lot of
/// solenoid control valves, spring loaded valves, and a differential pressure mechanism.
///
/// We don't want to simulate all of those little bits, so the functions of the actuator are split into
/// the following functional modes:
///
/// - [LinearActuatorMode.ClosedValves]: Turns actuator in a high constant spring/damper system simulating a closed actuator
/// only constrained by its own fluid compressibility.
/// - [ActiveDamping.LinearActuatorMode]: Actuator use internal valves to provide a force resisting to its own movements, dampening
/// the piece movements it's connected to.
/// - [LinearActuatorMode.PositionControl]: -> Actuator will try to use hydraulic pressure to move to a requested position, while
/// maintaining flow limitations.
```